taskcluster/mozillavpn_taskgraph/transforms/beetmover.py

```diff
@@ -125,79 +80,133 @@ def add_beetmover_worker_config(config, tasks):
+                        f"{app_version}-candidates",
+                        f"build{build_id}",
+                        build_os,
+                    )
+                )
+            elif phase == "ship-client":
+                assert build_os
+                destination_paths.append(
+                    os.path.join(
+                        "pub",
+                        "vpn",
+                        "releases",
+                        app_version,
+                        build_os,
+                    )
+                )
+
+            worker = deepcopy(worker_defaults)
+            worker["artifact-map"] = []
+            for artifact in upstream_artifacts:
+                worker["artifact-map"].append(
+                    {
+                        "taskId": artifact["taskId"],
+                        "paths": {
+                            path: {
+                                "destinations": [
+                                    os.path.join(
+                                        destination_path,
+                                        os.path.basename(path),
+                                    )
+                                    for destination_path in destination_paths
+                                ]
+                            }
+                            for path in artifact["paths"]
+                        },
+                    }
+                )
+
+            attributes = {
+                **task["attributes"],
+                "shipping-phase": phase,
+            }
+
+            dest = (
+                f"{archive_url}{destination_paths[0]}"
+                if destination_paths
+                else archive_url
+            )
+            if build_type == "addons/opt":
+                task_description = (
+                    f"This {worker_type} task will upload the {task['name']} to {dest}/"
+                )
+            elif phase == "ship-client":
+                task_description = f"This {worker_type} task will copy build {build_id} from candidates to releases"
+            else:
+                task_description = f"This {worker_type} task will upload a {build_os} release candidate for v{app_version} to {dest}/"
+
+            if phase.startswith("promote"):
+                worker["action"] = "push-to-candidates"
+            elif phase == "ship-addons":
+                worker["action"] = "direct-push-to-bucket"
+            elif phase == "ship-client":
+                worker["action"] = "push-to-releases"
+            else:
+                raise Exception(f"Invalid shipping_phase `{phase}`")
+
+            extra = {
+                "release_destinations": [
+                    f"{archive_url}{dest}/" for dest in destination_paths
+                ]
+            }
+            task_def = {
+                "name": f"{phase.split('-')[0]}-{task['name']}",
+                "description": task_description,
+                "dependencies": task["dependencies"],
```

Comment:
"push-to-release" beetmover tasks should depend on the corresponding "push-to-candidates" ones, since they're copying files from candidates to release.

Looking at this a bit more closely, instead of duplicating everything between promote and push we should probably  have a single push-to-release task instead of one per build.  It's already kind of broken today, e.g. looking at the log for https://firefox-ci-tc.services.mozilla.com/tasks/B9aYVb6VTLqzvqk0UME84g which is supposed to be "beetmover-android-arm64", it actually copied everything from `pub/vpn/candidates/2.19.0-candidates/build20231127060557/` to `pub/vpn/releases/2.19.0/`, regardless of platform, so we're basically copying these same files over and over, which is more confusing than helpful.